src/main.rs

```diff
@@ -54,30 +49,42 @@ fn run_target_program(elf_path: &Path, chip_name: &str, opts: &cli::Opts) -> any
     // connect to probe and flash firmware
     let probe_target = lookup_probe_target(elf_path, chip_name, opts)?;
     let mut sess = attach_to_probe(probe_target.clone(), opts)?;
-    flash(&mut sess, elf_path, opts)?;
+    if opts.reset {
+        flash(&mut sess, elf_path, opts)?;
+    }
 
-    // attack to core
+    // attach to core
     let memory_map = sess.target().memory_map.clone();
     let core = &mut sess.core(0)?;
 
-    // reset-halt the core; this is necessary for analyzing the vector table and
-    // painting the stack
-    core.reset_and_halt(TIMEOUT)?;
+    // reset if requested, and then halt the core; this is necessary for analyzing
+    // the vector table and painting the stack
+    if opts.reset {
+        core.reset_and_halt(TIMEOUT)?;
+    } else {
+        core.halt(TIMEOUT)?;
+    }
 
     // gather information
-    let (stack_start, reset_fn_address) = analyze_vector_table(core)?;
     let elf_bytes = fs::read(elf_path)?;
-    let elf = &Elf::parse(&elf_bytes, elf_path, reset_fn_address)?;
-    let target_info = TargetInfo::new(elf, memory_map, probe_target, stack_start)?;
-
-    // install stack canary
-    let canary = Canary::install(core, &target_info, elf, opts.measure_stack)?;
-    if opts.measure_stack && canary.is_none() {
-        bail!("failed to set up stack measurement");
+    let elf = &Elf::parse(&elf_bytes, elf_path)?;
+    let target_info = TargetInfo::new(elf, memory_map, probe_target)?;
+
+    let canary;
+    if opts.reset {
+        // install stack canary
+        canary = Canary::install(core, &target_info, elf, opts.measure_stack)?;
+        if opts.measure_stack && canary.is_none() {
+            bail!("failed to set up stack measurement");
+        }
+    } else {
+        // cannot safely touch the stack of a running application
+        canary = None;
+        log::warn!("You are using `--no-reset` -- stack overflow checks will not be done")
     }
 
     // run program and print logs until there is an exception
-    start_program(core, elf)?;
+    attach_to_program(core, elf)?;
```

Comment:
Gotta update it here too:
```suggestion
    attack_the_program(core, elf)?;
```

Reply:
Oh no 😆  I am sorry. This was meant to be a joke 🙈  Please change it back to `attach_to_program` 🤣